File: sdk/samples/FrameworkSamples/NETCF/Applications/RomanLegion/cs/romanlegion/barbarian.cs
//=====================================================================
//  File:      Barbarian.cs
//
//---------------------------------------------------------------------
//  This file is part of the Microsoft .NET Framework SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Drawing;
using System.Drawing.Imaging;
using System.Windows.Forms;
using System.Reflection;
using System.Threading;
using System.ComponentModel;

namespace Microsoft.Samples.RomanLegion
{
	public class Barbarian : BaseObj
	{
		private const int KeyboardMove = 5;

		private bool fMoveLeft;
		private bool fMoveRight;
		private bool fKeyboard;

		public Barbarian(Game game) : base(game)
		{
			//Barbarian Always Alive
			IsAlive = true;
		
			// Bmp Size
			Width = 14;
			Height = 10;
	
			// Start Position
			Left = MaximumX / 2;
			Top  = CurrentGame.WorldBounds.Height - Height - 20;
			LeftOld = Left;
			TopOld  = Top;

			// Load Bmp
			AnimationCells = new Bitmap[1];
			CellAttributes = new ImageAttributes[1];
			Load("Barbarian.bmp", 0);
		}

		public void Process()
		{
			Point ptMouse;
			LeftOld = Left;

			// If input is from Keyboard
			if (fKeyboard)
			{
				if (fMoveLeft)
					Left -= KeyboardMove;
				else if (fMoveRight)
					Left += KeyboardMove;
			}
			// If the mouse move
			else
			{
				ptMouse = CurrentGame.MouseLoc;
				Left = ptMouse.X - (Width / 2);
			}

			// Limit the Barbarian position to the window
			if (Left < 0)
				Left = 0;
			else if ((Left + Width) >= MaximumX)
				Left = MaximumX - Width;
		}
	
		// Set the Barbarian to move left
		public void GoLeft(bool move)
		{
            fKeyboard = true;
            fMoveLeft = move;
		}
	
		// set the Barbarian to move right
		public void GoRight(bool move)
		{
            fKeyboard = true;
            fMoveRight = move;
		}

		public void KeyboardOff()
		{
            fKeyboard = false;
		}
			
		public bool CollidesWith(Arrow arrow)
		{
			bool bCollide = false;

			// Check if the Arrow hits the Barbarian 
			if (this.Bounds.IntersectsWith(arrow.Bounds))
				bCollide = true;
			
            return(bCollide);
		}
	}
}

File: sdk/samples/FrameworkSamples/NETCF/Applications/RomanLegion/vb/romanlegion/barbarian.vb
'
' File: Barbarian.vb
'
'---------------------------------------------------------------------
' This file is part of the Microsoft .NET Framework SDK Code Samples.
'
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Namespace Microsoft.Samples.RomanLegion

   Public Class Barbarian
       Inherits BaseObj
   
       Private Const KeyboardMove As Integer = 5
   
       Private fMoveLeft As Boolean
       Private fMoveRight As Boolean
       Private fKeyboard As Boolean
   
       Public Sub New(ByVal game As Game)
           MyBase.New(game)
   
           'Barbarian Always Alive
           IsAlive = True
   
           ' Bmp Size
           CellWidth  = 14
           CellHeight = 10
   
           ' Start Position
           CellLeft = CInt(MaximumX / 2)
           CellTop  = CurrentGame.WorldBounds.Height - CellHeight - 20
           LeftOld = CellLeft
           TopOld  = CellTop
   
           ' Load Bmp
           ReDim AnimationCells(1)
           ReDim CellAttributes(1)
           Load("Barbarian.bmp", 0)
       End Sub
   
       Public Sub Process()
   
           Dim ptMouse As Point
           LeftOld = CellLeft
   
           ' If input is from Keyboard
           If (fKeyboard) Then
   
               If (fMoveLeft) Then
                   CellLeft -= KeyboardMove
               ElseIf (fMoveRight) Then
                   CellLeft += KeyboardMove
               End If
               ' If the mouse move
           Else
   
               ptMouse = CurrentGame.MouseLoc
               CellLeft = ptMouse.X - CInt(CellWidth / 2)
   
           End If
   
           ' Limit the Barbarian position to the window
           If (CellLeft < 0) Then
               CellLeft = 0
           ElseIf ((CellLeft + CellWidth) >= MaximumX) Then
               CellLeft = MaximumX - CellWidth
           End If
       End Sub
   
       ' Set the Barbarian to move left
       Public Sub GoLeft(move As Boolean)
           fKeyboard = True
           fMoveLeft = move
       End Sub
   
       ' set the Barbarian to move right
       Public Sub GoRight(move as Boolean)
           fKeyboard = True
           fMoveRight = move
       End Sub
   
       Public Sub KeyboardOff()
           fKeyboard = False
       End Sub
   
       Public Function CollidesWith(ByVal arrow As Arrow) As Boolean
   
           CollidesWith = False
           If (Me.Bounds.IntersectsWith(arrow.Bounds)) Then
               CollidesWith = True
           End If
           Return (CollidesWith)
       End Function
   End Class

End Namespace
   
